include/libtorrent/hasher512.hpp

```diff
@@ -112,7 +117,7 @@ namespace libtorrent
 		gcry_md_hd_t m_context;
 #elif TORRENT_USE_COMMONCRYPTO
 		CC_SHA512_CTX m_context;
-#elif TORRENT_USE_CRYPTOAPI
+#elif TORRENT_USE_CRYPTOAPI && defined(CALG_SHA_512)
```

Comment:
i.e. make config.hpp not set ``TORRENT_USE_CRYPTOAPI`` when building for a version that doesn't support it